src/main/java/org/broadinstitute/hellbender/tools/spark/sv/BreakpointPair.java

```diff
@@ -0,0 +1,476 @@
+    /**
+     * This function, when given overlaps of two corresponding regions on reference and contig sequences,
+     * attempts to find--naively and slowly--the copy numbers on the reference and the contig of tandem repeats,
+     * as well as the pseudo-homologyForwardStrandRep between the duplicated sequence and the right flanking region.
+     * The returned array of length 4 have entries in the following order:
+     *  estimate of copy number on the sequence where {@code overlapOnHigherCNSequence} was computed,
+     *  estimate of copy number on the sequence where {@code overlapOnLowerCNSequence} was computed,
+     *  duplicated unit length,
+     *  pseudo-homologyForwardStrandRep length
+     */
+    @VisibleForTesting
+    static int[] tandemDuplicationComplication(final int overlapOnHigherCNSequence, final int overlapOnLowerCNSequence) {
+
+        int higherCnEst=0, lowerCnEst=0, unitLen=0, pseudoHomLen=0;
+        double err = Double.MAX_VALUE;
+        for(int cn2=1; cn2<MAX_LOWER_CN; ++cn2) {
+            for(int cn1=cn2+1; cn1<=2*cn2; ++cn1) {
+                final int dupLenUpperBound = cn1==2*cn2 ? overlapOnLowerCNSequence : overlapOnHigherCNSequence;
+                for (int l=2; l<=dupLenUpperBound;++l) {
+                    for (int lambda=0; lambda<l; ++lambda) {
+                        final int d1 = (2*cn2 - cn1)*l + lambda;
+                        final int d2 = cn2*l + lambda;
+                        final double newErr = Math.abs(overlapOnHigherCNSequence-d1) + Math.abs(overlapOnLowerCNSequence-d2);
+                        if(newErr<err){
+                            err = newErr;
+                            higherCnEst = cn1; lowerCnEst = cn2;
+                            unitLen= l; pseudoHomLen = lambda;
+                        }
+                        if (err<1) return new int[]{lowerCnEst, higherCnEst, unitLen, pseudoHomLen};
+                    }
+                }
+            }
+        }
+
+        return new int[]{lowerCnEst, higherCnEst, unitLen, pseudoHomLen};
+    }
+
+    // TODO: 12/11/16 again this does not deal with inter-chromosome translocation yet
+    /**
+     * Returns the reference coordinates of the left and right breakpoints implied by this chimeric alignment.
+     * If there is homologous sequence represented in the alignments, it will be assigned to the side of the breakpoint
+     * with higher reference coordinates.
+     */
+    @VisibleForTesting
+    static Tuple2<SimpleInterval, SimpleInterval> leftJustifyBreakpoints(final ChimericAlignment ca, final BreakpointComplication complication) {
+
+        final int homologyLen = complication.homologyForwardStrandRep.length();
+
+        final String leftBreakpointRefContig, rightBreakpointRefContig;
+        final int leftBreakpointCoord, rightBreakpointCoord;
+        if(complication.refDup!=0 && complication.contigDup!=0) { // todo : development artifact-- assuming tandem duplication is not co-existing with inversion
```

Comment:
Yes. If you look at how the complications are resolved in `resolveComplications()` the distance between alignment regions logic is NOT triggered if there's strand switch, which is a must for inversion. So in inversion calling we are assuming insertions and deletions are no co-existing (inversion are still going to be annotated with inserted sequence and homology).